src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -192,197 +193,175 @@ QRegularExpression AutoDownloadRule::cachedRegex(const QString &expression, bool
     // The cache is cleared whenever the regex/wildcard, must or must not contain fields or
     // episode filter are modified.
     Q_ASSERT(!expression.isEmpty());
-    QRegularExpression regex(m_dataPtr->cachedRegexes[expression]);
 
-    if (!regex.pattern().isEmpty())
-        return regex;
+    QRegularExpression &regex {m_dataPtr->cachedRegexes[expression]};
+    if (regex.pattern().isEmpty()) {
+        regex = QRegularExpression {
```

Comment:
the new unnamed `QRegularExpression` object won't live after this statement, you shouldn't use reference to hold it.

Reply:
It's already reference.
But unfortunately gcc can't compile it with this syntax. I'll fix it later.

Reply:
you should use `QRegularExpression regex;` not `QRegularExpression &`

Reply:
You misunderstood something... it creates new item and returns reference to it.

Reply:
I think the `QRegularExpression` on this line is destroyed after going out of if block, then `regex` is referencing an invalid object, isn't it?

Reply:
No.
References are immutable. A reference is initialized only once, and then we refer to the object it points to. So here we just call the copy constructor.